### ISSUE TITLE
jsonschema2pojo-core v1.0.1 has CVE vulnerabilities on `scala-library-2.11.6` and `jackson-mapper-asl-1.9.13`

### ISSUE THREAD

**Issue by ariesmcrae** [joelittlejohn/jsonschema2pojo#1036]
I'm using owasp `dependency-check-maven` v5.2.2.

The `dependency-check-maven` is reporting that your `jsonschema2pojo-core v1.0.1` is using the following libraries with CVE vulnerabilities :

```
[ERROR] scala-library-2.11.6.jar: CVE-2017-15288

[ERROR] jackson-mapper-asl-1.9.13.jar: CVE-2017-15095, CVE-2017-17485, CVE-2017-7525, CVE-2018-1000873, CVE-2018-14718, CVE-2018-5968, CVE-2018-7489, CVE-2019-14540, CVE-2019-16335, CVE-2019-17267

```

Here's the mvn dependency:tree
```
[INFO] +- org.jsonschema2pojo:jsonschema2pojo-core:jar:1.0.1:compile
[INFO] |  +- org.scala-lang:scala-library:jar:2.11.6:compile
[INFO] |  +- org.codehaus.jackson:jackson-mapper-asl:jar:1.9.13:compile
```

Note:
I can upgrade `scala-library` on my POM file to the latest version, and that would make the CVE vulnerability to go away for that library only.

However, I can't upgrade `jackson-mapper-asl` because that has been deprecated its features have been been absorbed into the `jackson-databind`library.

**Comment by ddcruver** [comment]
@ariesmcrae the master branch is using:
org.scala-lang:scala-library:jar:2.11.12:compile
com.fasterxml.jackson.core:jackson-databind:jar:2.9.10:compile

**Comment by joelittlejohn** [comment]
We have Jackson 1.x support which is why we use jackson-mapper-asl 1.9.13. Short of dropping Jackson 1.x entirely I don't think we can really resolve this.

I'm not against dropping Jackson 1.x support. People can continue to use old versions of jsonschema2pojo if they need it.

**Comment by ddcruver** [comment]
Yeah dumb moment I referenced Jackson 2. Yeah since the last version of Jackson 1 dropping support is the only way to fix this overall.

In the mean time you could try to put an exclusion on the dependency in the plugin itself.

Dropping support for Jackson 1 since Jackson 2 has been out for almost 8 years doesn't seem like a bad idea for multiple reasons.  Jackson 1 hasn't been patched in almost 6 years.

**Comment by ddcruver** [comment]
@ariesmcrae you should check if org.scala-lang:scala-library:jar:2.11.12:compile is ok on the CVE front.

**Comment by ddcruver** [comment]
@joelittlejohn Another path would be to make the org.codehaus.jackson:jackson-mapper-asl an optional dependency, which would add a requirement that people add the dependency to the plugin if they require to use Jackson 1.  Regardless even if this route is chosen I believe Jackson 1 should be deprecated, not sure if that means a README excerpt or a log warning whenever it is used.

**Comment by joelittlejohn** [comment]
I think we should remove Jackson 1.x support, since it's no longer possible for people to use Jackson 1.x safely. Anyone still using it can use older versions of jsonschema2pojo.

**Comment by joelittlejohn** [comment]
And @ariesmcrae you should feel free to submit a PR that upgrades the scala libs.

**Comment by ddcruver** [comment]
@joelittlejohn Should the version be bumped higher?  Not sure if this deserves the 2.0 treatment or not.  But you could think of it this way jsonschema2pojo 2 would only supporting Jackson 2.

**Comment by joelittlejohn** [comment]
Yes, we'll bump to 1.1. This library basically follows 'Break Versioning' as roughly described here:
https://github.com/ptaoussanis/encore/blob/master/BREAK-VERSIONING.md
(although doing it long before Break Versioning was a thing :smile:) 

Since Jackson 1.x is now such a small niche (and really not safe to use since it is 'out of support') I'm happy to remove this in the change from 1.0 to 1.1.

**Comment by joelittlejohn** [comment]
I think we need to remember though that all these Jackson CVEs (for both versions of Jackson) actually don't affect jsonschema2pojo. They affect the systems that use Jackson (1 and 2) for their deserialization. We don't dictate what version of Jackson is used in the apps that use the types we generate, and people should use an up-to-date version there.

**Comment by ddcruver** [comment]
> Yes, we'll bump to 1.1. This library basically follows 'Break Versioning' as roughly described here:
> https://github.com/ptaoussanis/encore/blob/master/BREAK-VERSIONING.md
> (although doing it long before Break Versioning was a thing 😄)
> 
> Since Jackson 1.x is now such a small niche (and really not safe to use since it is 'out of support') I'm happy to remove this in the change from 1.0 to 1.1.

Interesting take on versioning; in the past I have tried to follow [Semantic Versioning 2.0.0](https://semver.org/)

**Comment by ddcruver** [comment]
> I think we need to remember though that all these Jackson CVEs (for both versions of Jackson) actually don't affect jsonschema2pojo. They affect the systems that use Jackson (1 and 2) for their deserialization. We don't dictate what version of Jackson is used in the apps that use the types we generate, and people should use an up-to-date version there.

Yeah we had to deal with these types of issues and it is all a matter if you system supports arbitrary class serialization.  The CVE thankfully don't say you can't use the library but dictate class they find that can be exploited.  Newer versions of Jackson Latest are released to address them by blacklisting the classes.

See example release notes: https://github.com/FasterXML/jackson/wiki/Jackson-Release-2.9.10

If you curious they do it here:
https://github.com/FasterXML/jackson-databind/blob/master/src/main/java/com/fasterxml/jackson/databind/jsontype/impl/SubTypeValidator.java

To sum up if we keep Jackson 2.x updated then it will avoid security libraries from flagging this module.

**Comment by ddcruver** [comment]
@ariesmcrae Also you should be able to override the version of Jackson 2 the plugin is using by adding the dependency in the plugin/plugin management section.

**Comment by joelittlejohn** [comment]
Superseded by #1200.
